libraries/aws_ssm_resource_compliance_summary.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsSsmResourceComplianceSummary < AwsResourceBase
+  name 'aws_ssm_resource_compliance_summary'
+  desc 'Verifies settings for a SSM Resource Compliance Summary'
+
+  example "
+    describe aws_ssm_resource_compliance_summary(resource_id: 'resource-id') do
+      it { should exist }
+    end
+  "
+
+  attr_reader :table
+
+  FilterTable.create
+             .register_column(:compliance_type,       field: :compliance_type)
+             .register_column(:resource_type,         field: :resource_type)
+             .register_column(:resource_id,           field: :resource_id)
+             .register_column(:status,                field: :status)
+             .register_column(:overall_severity,      field: :overall_severity)
+             .register_column(:execution_summary,     field: :execution_summary)
+             .register_column(:compliant_summary,     field: :compliant_summary)
+             .register_column(:non_compliant_summary, field: :non_compliant_summary)
+             .register_column(:compliance_item,       field: :compliance_item)
+             .install_filter_methods_on_resource(self, :table)
+
+  def initialize(opts = {})
+    opts = { resource_id: opts } if opts.is_a?(String)
+    super(opts)
+    validate_parameters(required: [:resource_id], allow: %i(compliance_type status))
+    @display_name = opts[:resource_id]
+
+    catch_aws_errors do
+      @compliance_summaries = return_resource_id_data(compliance_summary_data: @aws.ssm_client.list_resource_compliance_summaries)
```

Comment:
I think @omerdemirok 's comment above relates to this line e.g. you should be able to filter on the `resource_id` for the `@aws.ssm_client.list_resource_compliance_summaries` call if needed as per:
https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/SSM/Client.html#list_resource_compliance_summaries-instance_method if you need both method calls.  However, if there is significant overlap it might be worthwhile choosing only `aws_ssm_resource_compliance_summary` or `aws_ssm_resource_compliance_item` (or supporting both if more useful for the user).